src/ErgoDex/PContracts/PDeposit.hs

```diff
@@ -41,85 +44,89 @@ newtype DepositConfig (s :: S) = DepositConfig
-  txInfo        <- tletUnwrap $ hrecField @"txInfo" ctx
-  rewardPkh     <- tletUnwrap $ hrecField @"rewardPkh" datum
-  tokenA        <- tletUnwrap $ hrecField @"tokenA" datum
-  tokenB        <- tletUnwrap $ hrecField @"tokenB" datum
-  tokenLp       <- tletUnwrap $ hrecField @"tokenLp" datum
-  poolNft       <- tletUnwrap $ hrecField @"poolNft" datum
+depositValidatorT = plam $ \datum' redeemer' ctx' -> unTermCont $ do
+  ctx       <- tcont $ pletFields @'["txInfo", "purpose"] ctx'
+  datum     <- tcont $ pletFields @'["x", "y", "lq", "poolNft", "exFee", "rewardPkh", "collateralAda"] datum'
+  txInfo'   <- tletUnwrap $ hrecField @"txInfo" ctx
+  txInfo    <- tcont $ pletFields @'["inputs", "outputs", "signatories"] txInfo'
+  inputs    <- tletUnwrap $ hrecField @"inputs" txInfo
+  outputs   <- tletUnwrap $ hrecField @"outputs" txInfo
+
+  redeemer    <- tcont $ pletFields @'["poolInIx", "orderInIx", "rewardOutIx"] redeemer'
+  poolInIx    <- tletUnwrap $ hrecField @"poolInIx" redeemer
+  orderInIx   <- tletUnwrap $ hrecField @"orderInIx" redeemer
+  rewardOutIx <- tletUnwrap $ hrecField @"rewardOutIx" redeemer
+
+  rewardOut   <- tlet $ pelemAt # rewardOutIx # outputs
+  rewardPkh   <- tletUnwrap $ hrecField @"rewardPkh" datum
+  rewardValue <- tlet $ getRewardValue' # rewardOut # rewardPkh
+
+  poolIn'   <- tlet $ pelemAt # poolInIx # inputs
+  poolIn    <- tcont $ pletFields @'["outRef", "resolved"] poolIn'
+  poolValue <-
+    let pool = pfromData $ hrecField @"resolved" poolIn
+    in tletField @"value" pool
+  let
+    poolIdentity =
+      let
+        requiredNft = pfromData $ hrecField @"poolNft" datum
+        nftAmount   = assetClassValueOf # poolValue # requiredNft
+      in pif (nftAmount #== 1) (pcon PTrue) (pcon PFalse) 
+
+  selfIn'   <- tlet $ pelemAt # orderInIx # inputs
+  selfIn    <- tcont $ pletFields @'["outRef", "resolved"] selfIn'
+  selfValue <-
+    let self = pfromData $ hrecField @"resolved" poolIn
+    in tletField @"value" self
+
+  PSpending selfRef' <- tmatch (pfromData $ hrecField @"purpose" ctx)
+  let
+    selfIdentity =
+      let
+        selfRef   = pfromData $ pfield @"_0" # selfRef'
+        selfInRef = pfromData $ hrecField @"outRef" selfIn
+      in selfRef #== selfInRef
+
+  x  <- tletUnwrap $ hrecField @"x" datum
+  y  <- tletUnwrap $ hrecField @"y" datum
+  lq <- tletUnwrap $ hrecField @"lq" datum
+
   exFee         <- tletUnwrap $ hrecField @"exFee" datum
   collateralAda <- tletUnwrap $ hrecField @"collateralAda" datum
+
   let
-    validRefund  = hasValidSignatories # txInfo # rewardPkh
-    validDeposit = isValidDeposit # txInfo # poolNft # tokenA # tokenB # tokenLp # exFee # rewardPkh # collateralAda # redeemer
+    strictInputs =
+      let inputsLength = plength # inputs
+      in inputsLength #== 2
+    fairFee =
+      let outputAda = pGetLovelace # rewardValue
+      in collateralAda #<= outputAda
```

Comment:
will it work correctly in case when colAda is greater than minAdaValue and x|y is ada?